src/Compute/Compute/VirtualMachine/Config/NewAzureVMConfigCommand.cs

```diff
@@ -329,6 +351,49 @@ public override void ExecuteCmdlet()
                 vm.StorageProfile.ImageReference.Id = this.ImageReferenceId;
             }
 
+            if (this.IsParameterBound(c => c.SecurityType))
+            {
+                if (vm.SecurityProfile == null)
+                {
+                    vm.SecurityProfile = new SecurityProfile();
+                }
+                if (vm.SecurityProfile.UefiSettings == null)
+                {
+                    vm.SecurityProfile.UefiSettings = new UefiSettings();
+                }
+                vm.SecurityProfile.SecurityType = this.SecurityType;
+                if (vm.SecurityProfile.SecurityType == "TrustedLaunch" || vm.SecurityProfile.SecurityType == "ConfidentialVM")
+                {
+                    vm.SecurityProfile.UefiSettings.VTpmEnabled = vm.SecurityProfile.UefiSettings.VTpmEnabled == null ? true : this.EnableVtpm;
+                    vm.SecurityProfile.UefiSettings.SecureBootEnabled = vm.SecurityProfile.UefiSettings.SecureBootEnabled == null ? true : this.EnableSecureBoot;
+                }
+            }
+
+            if (this.IsParameterBound(c => c.EnableVtpm))
+            {
+                if (vm.SecurityProfile == null)
+                {
+                    vm.SecurityProfile = new SecurityProfile();
+                }
+                if (vm.SecurityProfile.UefiSettings == null)
+                {
+                    vm.SecurityProfile.UefiSettings = new UefiSettings();
+                }
+                vm.SecurityProfile.UefiSettings.VTpmEnabled = this.EnableVtpm;
+            }
+
+            if (this.IsParameterBound(c => c.EnableSecureBoot))
+            {
+                if (vm.SecurityProfile == null)
+                {
+                    vm.SecurityProfile = new SecurityProfile();
+                }
+                if (vm.SecurityProfile.UefiSettings == null)
+                {
+                    vm.SecurityProfile.UefiSettings = new UefiSettings();
+                }
+                vm.SecurityProfile.UefiSettings.SecureBootEnabled = this.EnableSecureBoot;
+
```

Comment:
you are missing a } for closing the if statement , this is causing a build error.

Reply:
Done thanks